src/guiapi/include/WindowMenuItems.hpp

```diff
@@ -42,14 +44,65 @@ class WI_SPIN_t : public AddSuper<IWiSpin> {
     T value;
     const T *range;
     const char *prt_format;
+    const char *units;
 
 protected:
-    virtual char *sn_prt() const;
+    virtual char *sn_prt() const override;
+    /* getMenuRects (not derived because of New/Old GUI and 3 rect array return (not 2))
+    *   Returns rect array according to this format: [label], [value], [units]
+    */
+    std::array<Rect16, 3> getSpinMenuRects(IWindowMenu &window_menu, Rect16 rect) const {
+        Rect16 base_rect = getRollingRect(window_menu, rect);
+        Rect16 unit_rect = { int16_t(base_rect.Left() + base_rect.Width() - 4 * window_menu.secondary_font->w), base_rect.Top(), uint16_t(4 * window_menu.secondary_font->w), base_rect.Height() };
+        base_rect -= unit_rect.Width();
+        char *buff = sn_prt();
+        Rect16 spin_rect = getCustomRect(window_menu, base_rect, strlen(buff) * window_menu.secondary_font->w);
+        base_rect -= spin_rect.Width();
+
+        return std::array<Rect16, 3> { base_rect, spin_rect, unit_rect };
+    }
+    /* derived printItem()
+    * Adition of new implementation:
+    *   values are offset from right (4 chars) and they are in secondary_font (smaller)
+    *   optional value units are added next to values (gray color)
+    */
+    virtual void printItem(IWindowMenu &window_menu, Rect16 rect, color_t color_text, color_t color_back, uint8_t swap) const override {
+#if (PRINTER_TYPE == PRINTER_PRUSA_MINI)
+        std::array<Rect16, 2> rects = IWiSpin::getMenuRects(window_menu, rect); // MINI GUI implementation
+#else                                                                           // PRINTER TYPE != PRUSA_PRINTER_MINI
+        std::array<Rect16, 3> rects = getSpinMenuRects(window_menu, rect);
+#endif                                                                          // PRINTER TYPE == PRUSA_PRINTER_MINI
+        //draw label
+        printLabel_into_rect(rects[0], color_text, color_back, window_menu.font, window_menu.padding, window_menu.GetAlignment());
+        //draw spin
+        // this MakeRAM is safe - temp_buff is allocated for the whole life of IWiSpin
+        render_text_align(rects[1], string_view_utf8::MakeRAM((const uint8_t *)temp_buff.data()),
+#if (PRINTER_TYPE == PRINTER_PRUSA_MINI)
```

Comment:
All these randomly scattered #ifdefs only show a poor design choice/principles. Remove them (use a derived class when some other functionality is needed)